node/core/pvf/src/host.rs

```diff
@@ -108,7 +118,10 @@ impl ValidationHost {
 	/// situations this function should return immediately.
 	///
 	/// Returns an error if the request cannot be sent to the validation host, i.e. if it shut down.
-	pub async fn heads_up(&mut self, active_pvfs: Vec<Pvf>) -> Result<(), String> {
+	pub async fn heads_up(
+		&mut self,
+		active_pvfs: Vec<(Pvf, ExecutorParams)>,
```

Comment:
nit: I wonder if `ExecutorParams` could be a part of the `Pvf` struct? Is that crazy? It would simplify a lot of the code here, and seems to make sense that we would keep the validation code closely associated with the corresponding EE params. At the very least, we could add a new type that joins them e.g. `PvfAndEnvironment`.

Reply:
> I wonder if `ExecutorParams` could be a part of the `Pvf` struct?

I believe it should not. In principle, a single PVF could be executed with different parameter sets, so `Pvf` abstracts a PVF itself and `ExecutorParams` abstracts a parameter set, such a design seems foolproof to me.

Yes, right now we have a situation that the execution parameters affect PVF code itself. But to me it seems more like a design bug, not a feature, introduced just because we don't have any better solution for gas metering than instrumenting the original WASM code. If we trade Wasmtime for WASMI, for example, that wouldn't be the case, and we could have exactly the same PVF code executed with different sets of parameters. So I prefer to keep them separated until a necessity of the opposite is proven.

> At the very least, we could add a new type that joins them e.g. `PvfAndEnvironment`.

And this one seems to be a legit idea. My original vector of tuples was probably a part of shitcoding marathon under the motto "make it work now and think of a good design later". That would also simplify interfacing with artifact abstractions which became somewhat ugly after the "artifact separation" commit. Thanks for pointing out, I'll think of that.